org-code-javabuilder/lib/src/main/java/org/code/neighborhood/GridSquare.java

```diff
@@ -0,0 +1,119 @@
+    private boolean passable;
+    private int paintCount;
+    private enum SquareType{ WALL, OPEN, START, FINISH, OBSTACLE, STARTANDFINISH, UNKNOWN};
+    private SquareType squareType;
+
+    public GridSquare(JSONObject descriptor) {
+        this.color = "";
+        try {
+            switch(Integer.parseInt(descriptor.get("tileType").toString())) {
+                case 0:
+                    this.squareType = SquareType.WALL;
+                    this.passable = false;
+                    break;
+                case 1:
+                    this.squareType = SquareType.OPEN;
+                    this.passable = true;
+                    break;
+                case 2:
+                    this.squareType = SquareType.START;
+                    this.passable = true;
+                    break;
+                case 3:
+                    this.squareType = SquareType.FINISH;
+                    this.passable = true;
+                    break;
+                case 4:
+                    this.squareType = SquareType.OBSTACLE;
+                    this.passable = false;
+                    break;
+                case 5:
+                    this.squareType = SquareType.STARTANDFINISH;
+                    this.passable = true;
+                    break;
+                default:
+                    this.squareType = SquareType.UNKNOWN;
+                    this.passable = false;
+            }
+        } catch (NumberFormatException e) {
+            System.out.println("\nPlease check the format of your grid.txt file, especially the tileType\n");
+            e.printStackTrace();
+            System.exit(1);
+        }
+        if(descriptor.containsKey("value")) {
+            try {
+                this.paintCount = Integer.parseInt(descriptor.get("value").toString());
+            } catch (NumberFormatException e) {
+                System.out.println("\nPlease check the format of your grid.txt file, especially the value\n");
+                e.printStackTrace();
+                System.exit(1);
+            }
+        } else {
+            this.paintCount = 0;
+        }
+    }
+
+    // Sets the color of the square to the given color
+    public void setColor(String color) {
+        if (color.length() > 1) {
+            System.out.println("use a 1 letter character to set your color");
+            return;
+        }
+        if (this.passable && this.paintCount == 0) {
+            this.color = color;
+        }
+    }
+
+    // Determines whether the given coordinate can be moved into
+    public boolean isPassable() {
+        return this.passable;
+    }
+
+    // Decreases the paintCount by 1 if there is available paint
+    public void collectPaint() {
+        if (this.containsPaint()) {
+            this.paintCount--;
+        } else {
+            System.out.println("There's no paint to collect here");
+        }
+    }
+
+    // Returns the square to a non-painted state
+    public void removePaint() {
```

Comment:
Is this function the inverse of setColor?  If so, maybe we can name it removeColor or unsetColor?